pkg/apis/infinispan/v2alpha1/backup_types.go

```diff
@@ -0,0 +1,96 @@
+package v2alpha1
+
+import (
+	v1 "github.com/infinispan/infinispan-operator/pkg/apis/infinispan/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// BackupSpec defines the desired state of Backup
+type BackupSpec struct {
+	Cluster   string                     `json:"cluster"`
+	Volume    BackupVolumeSpec           `json:"volume,optional,omitempty"`
+	Resources BackupResources            `json:"resources,optional,omitempty"`
+	Container v1.InfinispanContainerSpec `json:"container,optional,omitempty"`
+}
+
+type BackupVolumeSpec struct {
+	Storage          *string `json:"storage,optional,omitempty"`
+	StorageClassName *string `json:"storageClassName,optional,omitempty"`
+}
+
+type BackupResources struct {
+	Caches       []string `json:"caches,optional,omitempty"`
+	CacheConfigs []string `json:"cacheConfigs,optional,omitempty"`
+	Counters     []string `json:"counters,optional,omitempty"`
+	ProtoSchemas []string `json:"protoSchemas,optional,omitempty"`
+	Scripts      []string `json:"scripts,optional,omitempty"`
+}
+
+type BackupPhase string
```

Comment:
Phase should has an initial value in case of fail reconcile instead of empty value
```
  spec:
    cluster: example-infinispan
    path: asdasd
    volume:
      storage: 2Gi
  status:
    phase: ""
    pvc: example-infinispan-backup
```